crypto/src/merkle/proofs.rs

```diff
@@ -255,6 +255,172 @@ impl<H: Hasher> BatchMerkleProof<H> {
         v.remove(&1).ok_or(MerkleTreeError::InvalidProof)
     }
 
+    /// Computes the uncompressed Merkle paths which aggregate to this proof.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// * No indexes were provided (i.e., `indexes` is an empty slice).
+    /// * Number of provided indexes is greater than 255.
+    pub fn into_paths(&self, indexes: &[usize]) -> Result<Vec<Vec<H::Digest>>, MerkleTreeError> {
+        if indexes.is_empty() {
+            return Err(MerkleTreeError::TooFewLeafIndexes);
+        }
+        if indexes.len() > MAX_PATHS {
+            return Err(MerkleTreeError::TooManyLeafIndexes(
+                MAX_PATHS,
+                indexes.len(),
+            ));
+        }
+        let mut leaves = self.leaves.clone();
+        leaves.reverse();
+
+        let mut partial_tree = Vec::with_capacity(1 << (self.depth + 1));
+
+        for _ in 0..(1 << (self.depth + 1)) {
+            partial_tree.push(H::Digest::default());
+        }
```

Comment:
Could this be replaced with something like:
```Rust
let mut partial_tree = vec![H::Digest::default(), 1 << (self.depth + 1)];
```

Reply:
Indeed!